src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -918,19 +860,188 @@ export class SearchSource {
+      }),
     };
 
-    return omitByIsNil({ ...searchRequest, body: omitByIsNil(bodyToReturn) }) as SearchRequest;
+    return omitByIsNil({
+      ...searchRequest,
+      body: omitByIsNil(bodyToReturn),
+      indexType: this.getIndexType(index),
+      highlightAll:
+        searchRequest.highlightAll && builtQuery ? undefined : searchRequest.highlightAll,
+    }) as SearchRequest;
+  }
+
+  private getFieldFilter({
+    bodySourceExcludes,
+    metaFields,
+  }: {
+    bodySourceExcludes: any;
+    metaFields: any;
+  }) {
+    const filter = fieldWildcardFilter(bodySourceExcludes, metaFields);
+    return (fieldsToFilter: any) =>
+      fieldsToFilter.filter((fld: SearchFieldValue) => filter(this.getFieldName(fld)));
+  }
+
+  private getUniqueFieldNames({
+    fields,
+    fieldsFromSource,
+  }: {
+    fields: any;
+    fieldsFromSource: any;
+  }) {
+    const bodyFieldNames = fields.map((field: SearchFieldValue) => this.getFieldName(field));
+    return [...new Set([...bodyFieldNames, ...fieldsFromSource])];
+  }
+
+  private filterScriptFields({
+    uniqFieldNames,
+    scriptFields,
+  }: {
+    uniqFieldNames: any;
+    scriptFields: any;
+  }) {
+    return uniqFieldNames.includes('*')
+      ? scriptFields
+      : // filter down script_fields to only include items specified
+        pick(
+          scriptFields,
+          Object.keys(scriptFields).filter((f) => uniqFieldNames.includes(f))
+        );
+  }
+
+  private getBuiltEsQuery({
+    index,
+    query,
+    filters,
+    getConfig,
+    sort,
+  }: {
+    index: any;
+    query: any;
+    filters: any;
+    getConfig: (key: string) => any;
+    sort?: any;
+  }) {
+    // If sorting by _score, build queries in the "must" clause instead of "filter" clause to enable scoring
+    const filtersInMustClause = (sort ?? []).some((srt: EsQuerySortValue[]) =>
+      srt.hasOwnProperty('_score')
+    );
+    const esQueryConfigs = {
+      ...getEsQueryConfig({ get: getConfig }),
+      filtersInMustClause,
+    };
+    return buildEsQuery(index, query, filters, esQueryConfigs);
+  }
+
+  private getRemainingFields({
+    uniqFieldNames,
+    scriptFields,
+    runtimeFields,
+    _source,
+  }: {
+    uniqFieldNames: any;
+    scriptFields: any;
+    runtimeFields: any;
+    _source: any;
+  }) {
+    return difference(uniqFieldNames, [
+      ...Object.keys(scriptFields),
+      ...Object.keys(runtimeFields),
+    ]).filter((remainingField) => {
+      if (!remainingField) return false;
+      if (!_source || !_source.excludes) return true;
+      return !_source.excludes.includes(remainingField);
+    });
+  }
+
+  private getFieldsList({
+    index,
+    fields,
+    docvalueFields,
+    fieldsFromSource,
+    filteredDocvalueFields,
+    metaFields,
+    fieldListProvided,
+    sourceFieldsProvided,
+  }: {
+    index?: DataView;
+    fields: any;
+    docvalueFields: any;
+    fieldsFromSource: any;
+    filteredDocvalueFields: any;
+    metaFields: string;
```

Comment:
I don't think this could be a `string`, but it should be instead a `string[]` because of the schema associated with the UI settings of metafields: https://github.com/elastic/kibana/blob/dbc16cb897a0a3488b65e89406acb34c23b0d67a/src/plugins/data/server/ui_settings.ts#L39-L49